code/game/objects/structures/morgue.dm

```diff
@@ -35,41 +35,35 @@
 	update()
 
 /obj/structure/morgue/proc/update()
-	if(connected)
-		icon_state = "morgue0"
-		desc = initial(desc) + "\n[status_descriptors[1]]"
-	else
+	var/status
+	cut_overlays()
+	if(!connected)
 		if(contents.len)
-
 			var/mob/living/M = locate() in contents
-
 			var/obj/structure/closet/body_bag/B = locate() in contents
 			if(M==null) M = locate() in B
 
 			if(M)
 				var/mob/dead/observer/G = M.get_ghost()
-
 				if(M.mind && !M.mind.suicided)
 					if(M.client)
-						icon_state = "morgue3"
-						desc = initial(desc) + "\n[status_descriptors[4]]"
+						status = "revivable"
 					else if(G && G.client) //There is a ghost and it is connected to the server
-						icon_state = "morgue5"
-						desc = initial(desc) + "\n[status_descriptors[6]]"
+						status = "ghost"
 					else
-						icon_state = "morgue2"
-						desc = initial(desc) + "\n[status_descriptors[3]]"
+						status = "unrevivable"
 				else
-					icon_state = "morgue2"
-					desc = initial(desc) + "\n[status_descriptors[3]]"
-
-
+					status = "unrevivable"
 			else
-				icon_state = "morgue4"
-				desc = initial(desc) + "\n[status_descriptors[5]]"
+				status = "notbody"
 		else
-			icon_state = "morgue1"
-			desc = initial(desc) + "\n[status_descriptors[2]]"
+			status = "empty"
+		add_overlay("morgue_[status]")
+	else
+		status = "extended"
+	desc = initial(desc) + "\n[status_descriptors[status]]"
```

Comment:
I'd say leave the `desc` alone and instead override the `examine` proc to include the status description